### PR TITLE
Update and fix Measurement library

### PR DESCRIPTION
**Pull request by DGonzalezVillal**
Added a fix for the OVMF section part of the library and for the default launch policy.

Made better use of created types/structs and created missing instantiation functions for them.

### CODE REVIEW

**Review by larrydewey — DISMISSED** [review]
Looks pretty good to me. The changes suggested are not required.

File: src/measurement/gctx.rs [hunk 1]
```diff
@@ -80,17 +81,17 @@ impl Gctx<Updating> {
         page_info.extend_from_slice(&gpa.to_le_bytes());
 
         if page_info.len() != (page_info_len as usize) {
-            return Err(GCTXError::InvalidPageSize(
+            return Err(MeasurementError::GCTXError(GCTXError::InvalidPageSize(
```
Comment (larrydewey) [inline src/measurement/gctx.rs:84]:
```suggestion
            return Err(GCTXError::InvalidPageSize(...).into())
```

File: src/measurement/gctx.rs [hunk 2]
```diff
@@ -80,17 +81,17 @@ impl Gctx<Updating> {
...
                 page_info.len(),
                 page_info_len as usize,
-            ));
+            )));
         }
-        self.ld = sha384(&page_info);
+        self.ld = SnpLaunchDigest::try_from(sha384(&page_info).as_slice())?;
```
Comment (larrydewey) [inline src/measurement/gctx.rs:89]:
```suggestion
        self.ld = sha384(&page_info).as_slice().try_into()?;
```

File: src/measurement/gctx.rs [hunk 3]
```diff
@@ -116,7 +117,7 @@ impl Gctx<Updating> {
                     }
                     Ok(())
                 } else {
-                    Err(GCTXError::MissingData)
+                    Err(MeasurementError::GCTXError(GCTXError::MissingData))
```
Comment (larrydewey) [inline src/measurement/gctx.rs:120]:
```suggestion
                    Err(GCTXError::MissingData.into())
```

File: src/measurement/gctx.rs [hunk 4]
```diff
@@ -126,7 +127,7 @@ impl Gctx<Updating> {
                     self.update(page_type as u8, VMSA_GPA, sha384(data).as_slice())?;
                     Ok(())
                 } else {
-                    Err(GCTXError::MissingData)
+                    Err(MeasurementError::GCTXError(GCTXError::MissingData))
```
Comment (larrydewey) [inline src/measurement/gctx.rs:130]:
```suggestion
                    Err(GCTXError::MissingData.into())
```

File: src/measurement/gctx.rs [hunk 5]
```diff
@@ -140,7 +141,7 @@ impl Gctx<Updating> {
                     }
                     Ok(())
                 } else {
-                    Err(GCTXError::MissingBlockSize)
+                    Err(MeasurementError::GCTXError(GCTXError::MissingBlockSize))
```
Comment (larrydewey) [inline src/measurement/gctx.rs:144]:
```suggestion
                    Err(GCTXError::MissingBlockSize.into())
```

**Review by larrydewey — CHANGES_REQUESTED** [review]
Some minor changes, but overall looks good.

**Review by larrydewey — APPROVED** [review]
LGTM!

**Review by tylerfanelli — APPROVED** [review]

**Comment by tylerfanelli** [comment]
Closes #194

**Pull request merged**